solutions/USACO jan19-sleepy.mdx

```diff
@@ -0,0 +1,43 @@
+---
+id: usaco-892
+source: USACO Bronze 2019 January
+title: Sleepy Cow Sorting
+author: Ananth Kashyap
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_sleepy_bronze_jan19.html)
+
+In this problem, we are asked to find the minimum number of times steps before Farmer John will be able to completely sort the array of cows. We can think about this problem by looking at the end of the array of cows. Notice that if the last $i$ elements in the array are sorted in increasing order, since Famer John is acting optimally, he can fully sort the array in $n-i$ time steps. This is because, for the first $n-i$ numbers in the array, Farmer John can add them to space among the last $i$ elements such that they remain in increasing order, and notice that, after $n-i$ time steps, the array of cows will be completely sorted. 
```

Comment:
```suggestion
We can think about this problem by looking at the end of the array of cows. Notice that if the last $i$ elements in the array are sorted in increasing order, FJ can fully sort the cows in $n-i$ time steps. This is because the first $n-i$ elements are still to be sorted. Thus, we can find the last unsorted cow and output its position.
```

Reply:
You can either commit my suggestion or rewrite the whole thing.

Reply:
Also, I think that the official solution already explains this pretty well, so maybe you should delete this altogether.